247 fix/delete term meta

rajanvijayan opened the pull request:
Fix #247

rajanvijayan committed:
Add Delete Term Meta Module

Fix #247

rajanvijayan committed:
Apply fixes from StyleCI

rajanvijayan committed:
Merge pull request #357 from sudar/analysis-qojQv7

Apply fixes from StyleCI

rajanvijayan committed:
Add JS for fetch taxonomies
Fix #247

rajanvijayan committed:
Add delete term meta module

Fix #247

rajanvijayan committed:
Add equal and not equal options
Fix #247

rajanvijayan committed:
Update phpcs changes
Fix #247

rajanvijayan committed:
Update UI view
Fix #247

rajanvijayan committed:
Fix styleCI issue
Fix #247

rajanvijayan committed:
Remove empty space
Fix #247

rajanvijayan committed:
Apply fixes from StyleCI

rajanvijayan committed:
Merge pull request #452 from sudar/analysis-XkNlky

Apply fixes from StyleCI

rajanvijayan committed:
Merge remote-tracking branch 'origin/247-fix/delete-term-meta' into 247-fix/delete-term-meta

rajanvijayan committed:
Merge branch 'dev/6.0.0' into 247-fix/delete-term-meta

rajanvijayan committed:
Merge remote-tracking branch 'origin/247-fix/delete-term-meta' into 247-fix/delete-term-meta

rajanvijayan committed:
Update new metabox to module changes 
Fix #247

rajanvijayan committed:
Update else if issue 
Fix #247

rajanvijayan committed:
Update codecy issue : load_taxonomy_term_meta accesses the super-global variable $_GET.
Fix #247

rajanvijayan committed:
Revert global super variable issue 
Fix #247

rajanvijayan committed:
Add test case for delete term meta module 
Fix #350

rajanvijayan committed:
Merge branch 'dev/6.0.0' into 247-fix/delete-term-meta

rajanvijayan committed:
Update phpcs issue
Fix #247

rajanvijayan committed:
Add test cases for delete term custom taxonomies 
Fix #350

rajanit2000 commented:
Development is done, Waiting for review

sudar commented on assets/js/src/delete-term-meta.js:
```diff
@@ -0,0 +1,72 @@
+jQuery( document ).ready( function () {
+	
+	jQuery( 'input[name=smbd_meta_term_taxonomy]' ).change(function(){
+		jQuery( '.select2-terms' ).select2( {
```
Don't use the class name `select2`. It is causing issues.

Rename it to `enhanced-terms-dropdow`

rajanvijayan replied:
Fixed

sudar commented on include/Core/Base/Mixin/Fetcher.php:
```diff
@@ -22,6 +22,10 @@ protected function get_post_types() {
 		return bd_get_post_types();
 	}
 
+	protected function get_taxonomies() {
```
Add PHPDoc block comment for this method.

Basically if the method is not a inherited method then you have to add comments.

rajanvijayan replied:
Fixed

sudar commented on include/Core/Controller.php:
```diff
@@ -164,6 +165,31 @@ public function load_taxonomy_term() {
+	/**
+	 * Ajax call back function for getting taxonomies meta to load select2 options.
+	 *
+	 * @since 6.0.1
```
Change this to 6.0.0

rajanvijayan replied:
Fixed

sudar commented on include/Core/Controller.php:
```diff
@@ -164,6 +165,31 @@ public function load_taxonomy_term() {
+	public function load_taxonomy_term_meta() {
+		$response = array();
+
+		$term_id = sanitize_text_field( $_GET['term_id'] );
```
Use `absint` function instead of `sanitize_text_field`

rajanvijayan replied:
Fixed

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,176 @@
+/**
+ * Delete Term Meta.
+ *
+ * @since 6.1.0
```
Change to 6.0.0

rajanvijayan replied:
Fixed

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,176 @@
+	/**
+	 * Render the Modules.
+	 *
+	 * @return void
```
Don't add phpdoc block for inherited methods.

rajanvijayan replied:
Fixed

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,176 @@
+			<tr>
+				<td>
+					<select class="select2 select2-terms" name="smbd_<?php echo esc_attr( $this->field_slug ); ?>_term">
+						<option>Choose Terms</option>
```
This text should be made available for translation.

rajanvijayan replied:
Fixed

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,176 @@
+			</tr>
+		</table>
+
+		<h4><?php _e( 'Choose your term meta want to delete', 'bulk-delete' ); ?></h4>
```
Change to `Select the term meta that you want to delete`

rajanvijayan replied:
Fixed

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,176 @@
+			<tr>
+				<td>
+					<select class="select2 select2-term-meta" name="smbd_<?php echo esc_attr( $this->field_slug ); ?>_term_meta">
+						<option>Choose Term Meta</option>
```
Make this text available for translation by wrapping inside `_e` function. Do it for all other strings.

rajanvijayan replied:
Fixed

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,176 @@
+		 * Add more fields to the delete post meta field form.
+		 * This hook can be used to add more fields to the delete post meta field form.
+		 *
+		 * @since 5.4
```
Looks like you copy pasted and didn't change.

Replace post meta with term meta and change 5.4 to 6.0.0

rajanvijayan replied:
Fixed

sudar commented on include/ui/form.php:
```diff
@@ -198,6 +198,12 @@ function bd_get_post_types() {
 	return array_merge( $builtin_types, $custom_types );
 }
 
+function bd_get_taxonomies(){
```
If the function is going to only call `get_taxonomies` then directly use it. Don't create a new function for it.

rajanvijayan committed:
Fix select2 naming issue 
Fix #247

rajanvijayan committed:
Add PHP doc for get_taxonomies function 
Fix #247

rajanvijayan committed:
Update php doc and change milestone typo issue 
Fix #247

rajanvijayan committed:
Fix PHPCS issue
Fix #247

rajanvijayan committed:
Fix Milestone typo issue
Remove return void type
Add translation methods 
Update text message
Update action hook name 
Fix #247

rajanvijayan committed:
remove bd_get_taxonomies function 
Fix #247

rajanvijayan committed:
Add call direct function 
Fix #247

rajanit2000 commented on include/Core/Controller.php:
```diff
@@ -164,6 +165,31 @@ public function load_taxonomy_term() {
+	 *
+	 * @since 6.0.0
+	 */
+	public function load_taxonomy_term_meta() {
```
@sudar How can i fix this issue? I have fix all rest of changes

rajanvijayan committed:
Merge branch 'dev/6.0.0' into 247-fix/delete-term-meta

rajanvijayan committed:
Update function names
Add explicit checks
Fix #350

sudar committed:
Merge branch 'dev/6.0.0' into 247-fix/delete-term-meta

sudar commented on include/Core/Base/Mixin/Fetcher.php:
```diff
@@ -22,6 +22,17 @@ protected function get_post_types() {
+	 *
+	 * @return A list of taxonomy names
+	 */
+	protected function get_taxonomies() {
```
If you are only going to be calling `get_taxonomies` function, then there is no need for this function.

Remove it and use `get_taxonomies` directly.

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,174 @@
+		<h4><?php _e( 'Select the taxonomy whose term meta fields you want to delete', 'bulk-delete' ); ?></h4>
+		<table class="optiontable">
+		<?php
+		foreach ( $taxonomies as $taxonomy ) {
```
Use the `render_taxonomy_dropdown` method to render the taxonomies.

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,174 @@
+
+		</fieldset>
+
+		<p>
```
Use the `render_submit_button` helper function to get the submit button.

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,174 @@
+	protected function convert_user_input_to_options( $request, $options ) {
+		$options['term'] = esc_sql( bd_array_get( $request, 'smbd_' . $this->field_slug . '_term', 'term' ) );
+
+		$options['term_meta']       = bd_array_get( $request, 'smbd_' . $this->field_slug . '_term_meta', 'term_meta' );
```
Call `sanitize_text_field` function on the user input.

rajanvijayan replied:
@sudar how to use this method?

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,174 @@
+
+		if ( 'equal' === $options['term_meta_option'] ) {
+			$is_delete = delete_term_meta( $options['term'], $options['term_meta'], $options['term_meta_value'] );
+			if ( $is_delete ) {
```
Call this variable `$is_deleted`

sudar replied:
@rajanit2000 

The variable name should be `$is_deleted` not `$is_delete`

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,174 @@
+		} elseif ( 'not_equal' === $options['term_meta_option'] ) {
+			$term_value = get_term_meta( $options['term'], $options['term_meta'], true );
+			if ( $term_value !== $options['term_meta_value'] ) {
+				delete_term_meta( $options['term'], $options['term_meta'] );
```
Check the return value of `delete_term_meta` before incrementing the `count` variable.

rajanvijayan committed:
Update explicit test case with variable 
Fix #350

rajanvijayan committed:
Merge branch 'dev/6.0.0' into 247-fix/delete-term-meta

rajanvijayan committed:
Fix review changes
Fix #247

rajanvijayan committed:
Merge branch 'dev/6.0.0' into 247-fix/delete-term-meta

rajanvijayan committed:
Merge remote-tracking branch 'origin/247-fix/delete-term-meta' into 247-fix/delete-term-meta

rajanvijayan committed:
Fix same key multi value issue
Fix #247

rajanvijayan committed:
Apply fixes from StyleCI

rajanvijayan committed:
Merge pull request #488 from sudar/analysis-qgVPpd

Apply fixes from StyleCI

rajanvijayan committed:
Merge pull request #453 from sudar/350-fix/delete-term-meta-test

Add test case for delete term meta module

rajanvijayan committed:
Fix meta value issue 
Fix #247

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,167 @@
+	 */
+	public function filter_js_array( $js_array ) {
+		$js_array['dt_iterators'][]                 = '_' . $this->field_slug;
+		$js_array['validators']['delete_meta_term'] = 'noValidation';
```
Instead of hardcoding `delete_meta_term`, use `$this->action`

esudalaimadan replied:
Fixed.

sudar commented on include/Core/Metas/Modules/DeleteTermMetaModule.php:
```diff
@@ -0,0 +1,167 @@
+	 * @return string Success message.
+	 */
+	protected function get_success_message( $items_deleted ) {
+		/* translators: 1 Number of posts deleted */
```
Copy/paste error. Comment is not updated.

esudalaimadan replied:
Fixed.

sudar commented on tests/wp-unit/include/Core/Metas/Modules/DeleteTermMetaModuleTest.php:
```diff
@@ -0,0 +1,184 @@
+	/**
+	 * Add to test delete default taxonomy term meta with equal value.
+	 */
+	public function test_that_term_meta_can_be_deleted_with_default_taxonomy_in_equal_value() {
```
Rename to `test_that_term_meta_can_be_deleted_from_default_taxonomy_using_equal_operator.

Change the other function also to follow the correct format.

esudalaimadan replied:
Fixed.

sudar commented on tests/wp-unit/include/Core/Metas/Modules/DeleteTermMetaModuleTest.php:
```diff
@@ -0,0 +1,184 @@
+
+		$meta_deleted = $this->module->delete( $delete_options );
+
+		// Assert that post meta deleted.
```
Another copy/paste error. Check the comment.

esudalaimadan replied:
Fixed.

sudar commented:
@rajanit2000 

I am still seeing lot of copy/paste errors.

Are you doing a self review before submitting for PR?

sudar committed:
Merge branch 'dev/6.0.0' into 247-fix/delete-term-meta

sudar committed:
Apply fixes from StyleCI

sudar committed:
Merge pull request #526 from sudar/analysis-zD5Mpl

Apply fixes from StyleCI

sudar committed:
Merge branch 'dev/6.0.0' into 247-fix/delete-term-meta

sudar committed:
Merge branch 'dev/6.1.0' into 247-fix/delete-term-meta

sudar committed:
Merge branch 'dev/6.1.0' into 247-fix/delete-term-meta

esudalaimadan committed:
Cleanup module and tests.

Fix #247

esudalaimadan committed:
Merge branch 'dev/6.1.0' into 247-fix/delete-term-meta

sudar committed:
Merge branch 'dev/6.1.0' into 247-fix/delete-term-meta

esudalaimadan committed:
Merge branch 'dev/6.1.0' into 247-fix/delete-term-meta

sudar commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- include/Core/Metas/Modules/DeleteTermMetaModule.php  7
- assets/js/src/delete-term-meta.js  2
- tests/wp-unit/include/Core/Metas/Modules/DeleteTermMetaModuleTest.php  1
         

Clones added
============
- assets/js/src/delete-term-meta.js  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-delete/bulk-delete/pullRequest?prid=1939930&bid=8119203)

sudar commented on include/Core/Controller.php:
```diff
@@ -161,6 +162,31 @@ public function load_taxonomy_term() {
+	 *
+	 * @since 6.0.0
+	 */
+	public function load_taxonomy_term_meta() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [load_taxonomy_term_meta accesses the super-global variable $_GET.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=1939930)

sudar replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [load_taxonomy_term_meta accesses the super-global variable $_GET.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=1939930)

sudar committed:
Merge branch '247-fix/delete-term-meta' of github.com:sudar/bulk-delete into 247-fix/delete-term-meta

sudar commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- include/Core/Metas/Modules/DeleteTermMetaModule.php  7
- assets/js/src/delete-term-meta.js  2
- tests/wp-unit/include/Core/Metas/Modules/DeleteTermMetaModuleTest.php  1
         

Clones added
============
- assets/js/src/delete-term-meta.js  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-delete/bulk-delete/pullRequest?prid=1939930&bid=8119203)

sudar committed:
Change text labels

sudar commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- assets/js/src/delete-term-meta.js  2
- include/Core/Metas/Modules/DeleteTermMetaModule.php  7
- tests/wp-unit/include/Core/Metas/Modules/DeleteTermMetaModuleTest.php  1
         

Clones added
============
- assets/js/src/select2.js  2
- assets/js/src/delete-term-meta.js  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-delete/bulk-delete/pullRequest?prid=1939930&bid=8119203)

sudar commented on include/Core/Controller.php:
```diff
@@ -161,6 +162,31 @@ public function load_taxonomy_term() {
+	 *
+	 * @since 6.1.0
+	 */
+	public function load_term_metas() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [load_term_metas accesses the super-global variable $_GET.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=1939930)

sudar commented on include/Core/Controller.php:
```diff
@@ -131,11 +132,11 @@ public function verify_get_request_nonce( $result, $action ) {
 	 * @since 6.0.0
 	 */
-	public function load_taxonomy_term() {
+	public function load_taxonomy_terms() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [load_taxonomy_terms accesses the super-global variable $_GET.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=1939930)

sudar replied:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [load_taxonomy_terms accesses the super-global variable $_GET.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=1939930)

sudar committed:
Show all term meta without ajax

sudar commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 2
           

Complexity increasing per file
==============================
- assets/js/src/delete-term-meta.js  2
- include/Core/Metas/Modules/DeleteTermMetaModule.php  7
- tests/wp-unit/include/Core/Metas/Modules/DeleteTermMetaModuleTest.php  1
         

Clones added
============
- assets/js/src/select2.js  1
- assets/js/src/delete-term-meta.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/bulk-delete/bulk-delete/pullRequest?prid=1939930&bid=8119203)

sudar commented on include/Core/Controller.php:
```diff
@@ -161,6 +162,45 @@ public function load_taxonomy_term() {
+	 *
+	 * @since 6.1.0
+	 */
+	public function load_term_metas() {
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [load_term_metas accesses the super-global variable $_GET.](https://app.codacy.com/app/BulkWP/bulk-delete/pullRequest?prid=1939930)

sudar committed:
Use esc_attr_e intead of esc_attr

sudar committed:
Don't force long array syntax
